eng/pipelines/templates/steps/sync-repo.yml

```diff
@@ -0,0 +1,43 @@
+parameters:
+  # Provde a list of repos
+  # Example:
+  #   Azure/azure-rest-api-specs:
+  #     Branch: main
+  #     TargetRepos:
+  #       Azure/azure-rest-api-specs-pr:
+  #       azure-sdk/azure-rest-api-specs:
+  #         Branch: dev
+  #         Rebase: true
+  #   Azure/azure-sdk-for-go:
+  #     Branch: main
+  #     TargetRepos:
+  #       Azure/azure-sdk-for-go-pr:
+  - name: Repos
```

Comment:
I suggest having the GH_TOKEN parameter like the existing sync-repo-branch.yml so that the secret value can be passed in from the root instead of embedded here.

Reply:
In this comment https://github.com/Azure/azure-sdk-tools/pull/5479#issuecomment-1433861143 @benbp suggest use the PAT directly via the variable instead of passing it around via a github parameter. which one is better ?

Reply:
Sorry @JackTn let's go with Wes' approach here.

Reply:
🆗